lib/LLVMIRCodeGen/LLVMIRGen.cpp

```diff
@@ -545,8 +554,8 @@ llvm::Function *LLVMIRGen::getFunction(const std::string &name,
 /// the second BB is the loop exit BB.
 static std::pair<llvm::BasicBlock *, llvm::BasicBlock *>
 createLoop(llvm::IRBuilder<> &builder, llvm::LLVMContext &ctx,
-           llvm::Value *numElements) {
-  auto sizeTTy = builder.getIntNTy(sizeof(size_t) * 8);
+           llvm::Value *numElements, unsigned sizeTNumBits) {
```

Comment:
This `sizeTNumBits` parameter looks a bit ugly as it has very little to do with the semantics of `createLoop`. May be we should just make `createLoop` a method of LLVMIRGen? Then `createLoop` can call `getTargetSizeTWidth` when it needs to. WDYT?

Reply:
Sure, this makes sense to me.